Classes/Private/Manager/Submanagers/OCTSubmanagerCalls.m

```diff
@@ -199,56 +188,36 @@ - (BOOL)setVideoBitrate:(int)bitrate forCall:(OCTCall *)call error:(NSError **)e
 }
 
 #pragma mark Private methods
-- (OCTCall *)callFromFriend:(OCTToxFriendNumber)friendNumber
+- (OCTCall *)getOrCreateCallWithFriend:(OCTToxFriendNumber)friendNumber
 {
-    OCTDBManager *dbManager = [self.dataSource managerGetDBManager];
-    OCTDBChat *chatDB = [dbManager getOrCreateChatWithFriendNumber:friendNumber];
-    OCTChat *chat = [self.chatConverter objectFromRLMObject:chatDB];
-
-    OCTCall *call = [[OCTCall alloc] initCallWithChat:chat];
-
-    return call;
+    return [[self.dataSource managerGetRealmManager] getOrCreateCallWithFriendNumber:friendNumber];
 }
 
-- (void)startTimerForCall:(OCTCall *)call
+- (void)updateCall:(OCTCall *)call withStatus:(OCTCallStatus)status
 {
-    [self.calls updateCall:call updateBlock:^(OCTCall *callToUpdate) {
-        callToUpdate.status = OCTCallStatusInSession;
-        [callToUpdate startTimer];
+    OCTRealmManager *realmManager = [self.dataSource managerGetRealmManager];
+    [realmManager updateObject:call withBlock:^(id theObject) {
+        OCTCall *dbCall = theObject;
```

Comment:
You can use `OCTCall` type directly in block description:
```
[realmManager updateObject:call withBlock:^(OCTCall *theCall) { 
```

Also no `db` anymore ;-)